pkg/monitoring/feed_monitor.go

```diff
@@ -62,59 +67,88 @@ func (f *feedMonitor) Start(ctx context.Context) {
-				f.feedConfig.FeedName, f.feedConfig.FeedPath, f.solanaConfig.NetworkID,
-				f.solanaConfig.NetworkName)
-			isLateAnswer := time.Since(time.Unix(int64(typed.Answer.Timestamp), 0)).Seconds() > float64(f.feedConfig.HeartbeatSec)
-			f.metrics.SetOffchainAggregatorAnswerStalled(isLateAnswer, f.feedConfig.ContractAddress.String(),
-				f.solanaConfig.ChainID, f.feedConfig.ContractStatus, f.feedConfig.ContractType,
-				f.feedConfig.FeedName, f.feedConfig.FeedPath, f.solanaConfig.NetworkID,
-				f.solanaConfig.NetworkName)
-		}
 	}
 }
+
+func (f *feedMonitor) processState(envelope StateEnvelope) error {
+	var mapping map[string]interface{}
+	mapping, err := MakeConfigSetMapping(envelope, f.solanaConfig, f.feedConfig)
+	if err != nil {
+		return fmt.Errorf("failed to map message %v: %w", envelope, err)
+	}
+	value, err := f.stateSchema.Encode(mapping)
+	if err != nil {
+		return fmt.Errorf("failed to enconde message %v: %w", envelope, err)
+	}
+	var key = f.feedConfig.StateAccount.Bytes()
+	if err = f.envelopeProducer.Produce(key, value); err != nil {
+		return fmt.Errorf("failed to publish message %v: %w", envelope, err)
+	}
+	return nil
+}
+
+func (f *feedMonitor) processTransmission(envelope TransmissionEnvelope) error {
+	var mapping map[string]interface{}
+	mapping, err := MakeTransmissionMapping(envelope, f.solanaConfig, f.feedConfig)
+	if err != nil {
+		return fmt.Errorf("failed to map message %v: %w", envelope, err)
+	}
+	value, err := f.transmissionSchema.Encode(mapping)
+	if err != nil {
+		return fmt.Errorf("failed to enconde message %v: %w", envelope, err)
+	}
+	var key = f.feedConfig.StateAccount.Bytes()
+	if err = f.envelopeProducer.Produce(key, value); err != nil {
+		return fmt.Errorf("failed to publish message %v: %w", envelope, err)
+	}
+	f.metrics.SetHeadTrackerCurrentHead(envelope.BlockNumber, f.solanaConfig.NetworkName,
+		f.solanaConfig.ChainID, f.solanaConfig.NetworkID)
+	f.metrics.SetOffchainAggregatorAnswers(envelope.Answer.Data, f.feedConfig.ContractAddress.String(),
+		f.solanaConfig.ChainID, f.feedConfig.ContractStatus, f.feedConfig.ContractType,
+		f.feedConfig.FeedName, f.feedConfig.FeedPath, f.solanaConfig.NetworkID,
+		f.solanaConfig.NetworkName)
+	f.metrics.IncOffchainAggregatorAnswersTotal(f.feedConfig.ContractAddress.String(),
+		f.solanaConfig.ChainID, f.feedConfig.ContractStatus, f.feedConfig.ContractType,
+		f.feedConfig.FeedName, f.feedConfig.FeedPath, f.solanaConfig.NetworkID,
+		f.solanaConfig.NetworkName)
+	isLateAnswer := time.Since(time.Unix(int64(envelope.Answer.Timestamp), 0)).Seconds() > float64(f.feedConfig.HeartbeatSec)
+	f.metrics.SetOffchainAggregatorAnswerStalled(isLateAnswer, f.feedConfig.ContractAddress.String(),
+		f.solanaConfig.ChainID, f.feedConfig.ContractStatus, f.feedConfig.ContractType,
+		f.feedConfig.FeedName, f.feedConfig.FeedPath, f.solanaConfig.NetworkID,
+		f.solanaConfig.NetworkName)
+	return nil
+}
+
+func (f *feedMonitor) processTelemetry(envelope StateEnvelope) error {
```

Comment:
As I understand it, this is not telemetry, in the chainlink-ocr2 sense, no?! Telemetry goes into the implementation of `types.MonitoringEndpoint`. Please rename this helper